great_expectations/expectations/metrics/map_metric_provider.py

```diff
@@ -1669,6 +1900,34 @@ def _register_metric_functions(cls):
                             metric_provider=_sqlalchemy_column_map_condition_value_counts,
                             metric_fn_type=MetricFunctionTypes.VALUE,
                         )
+                    elif domain_type == MetricDomainTypes.MULTICOLUMN:
+                        register_metric(
+                            metric_name=metric_name + ".unexpected_count",
+                            metric_domain_keys=metric_domain_keys,
+                            metric_value_keys=metric_value_keys,
+                            execution_engine=engine,
+                            metric_class=cls,
+                            metric_provider=_sqlalchemy_map_condition_unexpected_count,
```

Comment:
@NathanFarmer Do we need this?  I notice that we already register the `unexpected_count` metric earlier.  If you agree, then we can delete the `_sqlalchemy_map_condition_unexpected_count()` helper method.  Happy to discuss.

Reply:
Without helper `_sqlalchemy_map_condition_unexpected_count()` and registering this `unexpected_count` in this block I get: `great_expectations.exceptions.exceptions.MetricProviderError: No provider found for multicolumn_sum.equal.unexpected_count using SqlAlchemyExecutionEngine`.

Reply:
If I remove the registration for the `unexpected_count` that starts at line 1862 and has `metric_provider=None`, `pytest` starts failing with various `KeyError`s for `result["unexpected_list"]`.

Reply:
@alexsherstinsky Following our offline conversation I have removed the duplicate helper and changed the `metric_provider` to `_sqlalchemy_map_condition_unexpected_count_value`.